### PR TITLE
Refactor repair request edit flow to side sheet

### PR DESCRIPTION
**Pull request by thienchi2109**
## Summary
- Refactor repair request edit UI from centered dialog to shared side-sheet shell.
- Extract shared repair request form fields/actions for create and edit flows.
- Rename row approve action to Xác nhận and add lucide icons to row action menu items.

## Test Plan
- ./node_modules/.bin/prettier --check --ignore-unknown changed repair-request files
- node scripts/npm-run.js run verify:no-explicit-any
- node scripts/npm-run.js run verify:dedupe
- node scripts/npm-run.js run typecheck
- node scripts/npm-run.js run test:run -- focused repair-request tests
- node scripts/npm-run.js run react-doctor

---
## Summary by cubic
Moved the repair request edit flow into the shared side sheet and unified create/edit with shared fields and actions. Clarified row action labels with icons and tightened desired-date validation; expanded tests to cover field edge cases.

- **Refactors**
  - Edit now uses `SideSheetShell`; submits via the existing update mutation and closes with the sheet.
  - Added `RepairRequestsFormFields` and `RepairRequestsSheetActions`; `RepairRequestsCreateSheetForm` now reuses them and removed `RepairRequestsCreateSheetActions`.
  - Row menu in `RepairRequestsColumns`: renamed "Duyệt" to "Xác nhận" and added icons for view, edit, delete, approve, and completion actions.
  - Edit desired date is now constrained to be on/after the request date.
  - Tests: added `RepairRequestsEditSheet.test.tsx` and expanded `RepairRequestsFormFields.test.tsx` for labels, permissions, and required-field edge cases; updated `RepairRequestRowActions.test.tsx` to assert the new label and icons.

<sup>Written for commit 7c7129a9dc68f3601b209792b92e9cefe07165cd. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/thienchi2109/qltbyt-nam-phong/pull/656?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Repair request forms now share a consistent layout, including optional desired completion date and repair unit/external company fields when applicable.
  * Repair request editing is now presented in a side sheet with clearer submit/cancel behavior.
  * Repair request row actions now use improved confirmation/approval wording and status icons.

* **Bug Fixes**
  * Improved form submission reliability and automatic closing after saving changes.
  * Enhanced table rendering for device/date/status details to make review and completion states easier to understand.

### CODE REVIEW

File: src/app/(app)/repair-requests/_components/RepairRequestsFormFields.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          id={repairItemsId}
+          placeholder="VD: Thay màn hình, sửa nguồn…"
+          rows={3}
+          value={repairItems}
+          onChange={(event) => onRepairItemsChange(event.target.value)}
+          required={repairItemsRequired}
+        />
+      </div>
+      <div className="space-y-2">
+        <Label>Ngày mong muốn hoàn thành (nếu có)</Label>
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/(app)/repair-requests/_components/RepairRequestsFormFields.tsx]: Fixed in a6f6a894: wired the desired-date label to the popover trigger via htmlFor/id and added focused form-field coverage.

File: src/app/(app)/repair-requests/__tests__/RepairRequestsEditSheet.test.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    isPending: false,
+  },
+  canSetRepairUnit: true,
+}
+
+vi.mock("../_hooks/useRepairRequestsContext", () => ({
+  useRepairRequestsContext: () => contextValue,
+}))
+
+vi.mock("@/components/shared/SideSheetShell", () => ({
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/(app)/repair-requests/__tests__/RepairRequestsEditSheet.test.tsx:60]: Fixed in a6f6a894: the SideSheetShell test mock now consumes open and the test asserts the controlled open state.
